bwapi/bwresources.py

```diff
@@ -50,34 +50,76 @@ def reload(self):
+                    raise ValueError('Could not find the resource ID {} in the project'.format(resource))
+                elif check == True:
+                    return False
+            pk = resource 
+        elif isinstance(resource, str):
+            entries = [pk for pk, name in self.names.items() if name == resource]
+            if len(entries) > 1:
+                #check status doesn't matter here - it's an error either way
+                raise ValueError('The resource name {} is ambiguous: {}'.format(resource, entries))
+            if entries:
+                if check == False:
+                    return entries[0]
+                elif check == True:
+                    return True
+            else:
+                try: 
+                    pk = int(resource)
+                except:
+                    if check == False:
+                        raise ValueError('Could not find the resource name {} in the project'.format(resource))
+                    elif check == True:
+                        return False
+        if pk not in self.names.keys():
+            if check == False:
+                raise ValueError('Could not find the resource ID {} in the project'.format(resource))
+            elif check == True:
+                return False
+        if pk:
+            if check == False:
+                return pk
+            elif check == True:
+                return True
+    
     def get(self, name=None):
         """
-        If you specify a name, this function will retrieve all information for that resource as it is stored in Brandwatch.
-        If you do not specify a name, this function will retrieve all information for all resources of that type as they are stored in Brandwatch.
+        If you specify an ID, this function will retrieve all information for that resource as it is stored in Brandwatch.
+        If you specify a name, this will be mapped to the appropriate ID. An error will be raised if there are two IDs with the name specified.
+        If you do not pass anything in with the `name` argument, this function will retrieve all information for all resources of that type as they are stored in Brandwatch.
 
         Args:
-            name:   Name of the resource that you'd like to retrieve - Optional.  If you do not specify a name, all resources of that type will be retrieved.
+            name: ID or name of the resource that you'd like to retrieve - Optional.  If you do not specify an ID, all resources of that type will be retrieved.
 
         Raises:
-            KeyError:   If you specify a resource name and the resource does not exist.
+            KeyError:   If you specify a resource ID and the resource does not exist.
 
         Returns:
             All information for the specified resource, or a list of information on every resource of that type in the account.
         """
-        if not name:
-            return self.project.get(endpoint=self.general_endpoint)["results"]
```

Comment:
Does the `name=None` behaviour still work? You seem to have deleted this

Reply:
The new get() and get_resource_id() methods handles this by get_resource_id() returning an empty string when no resource is provided to it, which means that get() retrieves the same information as it would with these lines.

I tested this like so:
```
r_old = queries.project.get(endpoint=queries.general_endpoint)
r_new = queries.get()
r_old == r_new
```
out[]: True